packages/ai/src/ai/account/store_providers/s3.py

```
@@ -308,6 +308,147 @@ async def list_files(self, prefix: str = '') -> list:
+            upload_id = response['UploadId']
+            return {
+                'context': {
+                    'key': key,
+                    'upload_id': upload_id,
+                    'parts': [],
+                    'part_number': 1,
+                    'buffer': bytearray(),
+                },
+            }
+        except Exception as e:
+            raise StorageError(f'Failed to start multipart upload for {filename}: {e}') from e
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)